### DISCUSSION TITLE
SQL Timeout Serverless

### DISCUSSION THREAD

**Discussion by szympulka** [AzureCommunityPL/group#450] [Pomoc]
Aplikacja łączy się do bazy danych (General Purpose - Serverless: Standard-series (Gen5), 4 vCores) przez connetion stringa, gdzie mam ustawiony timeout na 240s (";Connection Timeout=240;") requesty są zabijane po 30s. W dokumentacji nie ma nic opisane aby baza tak reagowała. Przy wykorzystaniu DTU, wszystko działało poprawnie. Spotkał się ktoś z Was z takim problemem?

**Comment by kamil-mrzyglod** [comment]
Masz na myśli:
* pierwszy request do SQL serverless -> 30s timeout -> koniec 
* pierwszy request do SQL serverless -> 30s timeout -> [retry policy, które nic nie daje] -> koniec

Connection Timeout jest po Twojej stronie więc baza niekoniecznie musi respektować to ustawienie.

**Reply by szympulka** [reply]
No właśnie to jest problem, bo nie respektuje tego w connetion stringu. Narzucenie timeout poprzez EntityFramework zadziałało. Pyttanie tylko, czemu na servellers się tak dzieje a na DTU nie?
 
Mam na myśli, że pierwszy request jest ubijany po 30s

**Reply by kamil-mrzyglod** [reply]
Musisz dać więcej informacji bo nadal nie czuję tego scenariusza:
* z czego korzystasz jak łączysz się z bazą (ADO / Dapper / EF / inne)
* gdzie dokładnie dostajesz błąd i jaki to błąd
* czy obsługujesz zerwanie pierwszego połączenia poprzez jakieś retry policy

Pytanie dodatkowe czy to przypadkiem nie jest Twój case:
> If a serverless database is paused, then the first login will resume the database and return an error stating that the database is unavailable with error code 40613. Once the database is resumed, the login must be retried to establish connectivity. Database clients with connection retry logic should not need to be modified. For connection retry logic options that are built-in to the SqlClient driver, see [configurable retry logic in SqlClient](https://docs.microsoft.com/en-us/sql/connect/ado-net/configurable-retry-logic).

**Comment by MariuszKrzanowski** [comment]
Wydaje mi się, że tutaj masz chyba problem z CommandTimeout, który ma domyślną wartość 30 sekund i to należy ustawiać przy każdym żądaniu/zapytaniu do SQL. Z tego co pamiętam, to ConnectionTimeout określa jak długo klient czeka na uzyskanie połączenia do serwera zanim zgłosi porażkę. I nie powinno to mieć wpływu na to jak długo to połączenie żyje. Chyba, że na SQL serwerze masz ustawione ubijanie IdleConnections, albo jakiś sidecar ma ubijanie połączeń trwających dłużej niż 30 sekund.

Mariusz
